recipes/openssh/all/conandata.yml

```diff
@@ -0,0 +1,12 @@
+sources:
+  "9.1p1":
+    url: "https://cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-9.1p1.tar.gz"
+    sha256: "19f85009c7e3e23787f0236fbb1578392ab4d4bf9f8ec5fe6bc1cd7e8bfdd288"
+  "8.1p1":
+    url: "https://cdn.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-8.1p1.tar.gz"
+    sha256: "02f5dbef3835d0753556f973cd57b4c19b6b1f6cd24c03445e23ac77ca1b93ff"
+patches:
+  "9.1p1-neutrino-7.0":
```

Comment:
This needs to be changed. You should add a `patch_system: "neutrino-7.0"` and filter by that key instead. Else, changes in the patches for an unrelated version will change the revisions of every other version!

Reply:
So what you mean is that I should apply this filtering when copying in `export_sources(...)` in order to only export patches for the package I am building?

Reply:
No, my above suggestion cannot be true as sources must be configuration independent.
Can you elaborate on what you mean? 
How can this happen: `changes in the patches for an unrelated version will change the revisions of every other version!`?

Reply:
@RubenRBS I have committed a patch solution and added os and os_version keys to conan_data. Is that what you were looking for?

Reply:
That, unfortunately, failed due to a hook preventing patch data having a `patch_os` and `patch_os_version` key

Reply:
Agghh, let me check with the team to see if taking care of the hook is the way to go, thanks for your patience in this

Reply:
@RubenRBS Any news?

Reply:
@RubenRBS Any news, or should I go another route implementing this?

Regards
Klaus

Reply:
Hi, sorry for the delay, thank you a lot for your patience - This is similar to an issue raised in https://github.com/conan-io/conan-center-index/pull/19529, which we still need to come to a conclusion on.

Note that @uilianries has started work on the linters migration, so they can probably chime in if they think it would be faster to move away from this implementation, or else wait for the new linters to allow for this

Reply:
Hi @klausholstjacobsen disregard my last message, we've come to the conclusion that this is something worth supporting, will make sure the linter does not complain for now, thanks again for your patience :)